TESTING.md

```diff
@@ -1,89 +1,483 @@
+* Restricted capabilities are applied to a running container
+* Valuable CLI checks (when available)
+
+[Kong](https://github.com/bitnami/charts/blob/main/.vib/kong/goss/goss.yaml) or [MetalLB](https://github.com/bitnami/charts/blob/main/.vib/metallb/goss/goss.yaml) are two good examples of tests implementing some of the above.
+
+### Run it locally
+
+Sometimes it is of interest to run the tests locally, for example during development. Though there may be different approaches, you may follow the steps below to execute the tests locally:
+
+1. Deploy the target Chart in your cluster, using the same installation parameters specified in the `vib-verify.json` pipeline file
+
+    ```bash
+    $ cat .vib/nginx/vib-verify.json | grep "runtime_parameters"
+            "runtime_parameters": "Y29udGFpbmVyUG9ydHM6CiAgaHR0cDogODA4MQogIGh0dHBzOiA4NDQ0CnBvZFNlY3VyaXR5Q29udGV4dDoKICBlbmFibGVkOiB0cnVlCiAgZnNHcm91cDogMTAwMgpjb250YWluZXJTZWN1cml0eUNvbnRleHQ6CiAgZW5hYmxlZDogdHJ1ZQogIHJ1bkFzVXNlcjogMTAwMgpzZXJ2aWNlOgogIHR5cGU6IExvYWRCYWxhbmNlcgogIHBvcnRzOgogICAgaHR0cDogODAKICAgIGh0dHBzOiA0NDQK",
+    $ helm install nginx bitnami/nginx -f <(echo "Y29udGFpbmVyUG9ydHM6CiAgaHR0cDogODA4MQogIGh0dHBzOiA4NDQ0CnBvZFNlY3VyaXR5Q29udGV4dDoKICBlbmFibGVkOiB0cnVlCiAgZnNHcm91cDogMTAwMgpjb250YWluZXJTZWN1cml0eUNvbnRleHQ6CiAgZW5hYmxlZDogdHJ1ZQogIHJ1bkFzVXNlcjogMTAwMgpzZXJ2aWNlOgogIHR5cGU6IExvYWRCYWxhbmNlcgogIHBvcnRzOgogICAgaHR0cDogODAKICAgIGh0dHBzOiA0NDQK" | base64 -d)
+    ```
+
+1. Download the [GOSS binary for Linux AMD64](https://github.com/goss-org/goss/releases/)
+1. Copy the binary and test files to the target pod where it should be executed
+
+    ```bash
+    $ kubectl get pods
+    NAME                    READY   STATUS    RESTARTS   AGE
+    nginx-5fbc8786f-95rpl   1/1     Running   0          17m
+
+    $ kubectl cp ./goss-linux-amd64 nginx-5fbc8786f-95rpl:/tmp/
+    $ kubectl cp .vib/nginx/goss/goss.yaml nginx-5fbc8786f-95rpl:/tmp/
+    $ kubectl cp .vib/nginx/goss/vars.yaml nginx-5fbc8786f-95rpl:/tmp/
+    ```
+1. Grant execution permissions to the binary and launch the tests
+
+    ```bash
+    $ kubectl exec -it nginx-5fbc8786f-95rpl -- chmod +x /tmp/goss-linux-amd64
+    $ kubectl exec -it nginx-5fbc8786f-95rpl -- /tmp/goss-linux-amd64 --gossfile /tmp/goss.yaml --vars /tmp/vars.yaml validate
+    .........
+
+    Total Duration: 0.011s
+    Count: 9, Failed: 0, Skipped: 0
+    ```
+
+### Specific acceptance criteria
+
+* [ ] Main test file name should be `goss.yaml`
+* [ ] Deployment-related parameters should be specified in a file named `vars.yaml`. This is a subset of the `runtime_parameters`
+* [ ] Use templating to parametrize tests with the help of the `vars.yaml` file
+* [ ] Tests should not rely on system packages (e.g. `curl`). Favor built-in GOSS primitives instead
+* [ ] Only check paths and configuration files that are injected/used by the Helm Chart (configmap, PVC, ...). Do not check the ones already present in the container
```

Comment:
I understand what you want to say, but I think this can be misleading. As you noted above, we sometimes need to check config files are using the values defined at `runtime_parameters`.